### PR TITLE
Fixed identified bugs from PE dry run

### PR DESCRIPTION
**Pull request by Dineshraj555**

### CODE REVIEW

**Review by Tiffanylin21 — COMMENTED** [review]
lgtm!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -144,19 +145,18 @@ public ReadOnlyAddressBook undo() throws Exception {
     public void updateFilteredPetList(Predicate<Pet> predicate) {
         requireNonNull(predicate);
         this.lastUsedPredicate = predicate;
-        filteredPets.setPredicate(predicate);
+        filteredPets.setPredicate(lastUsedPredicate);
```
Comment (Tiffanylin21) [inline src/main/java/seedu/address/model/ModelManager.java:148]: Perhaps this one should be `filteredPets.setPredicate(predicate);` instead?

**Comment by tobihy** [comment]
For the test cases that are currently not passing, do update the `expectedModel` under the respective test cases so that they follow the new behaviour of the pet list, i.e. filter them by previous predicate. 

The reason the test cases are failing is that previously, after these commands, the `expectedModel` would be the list of **all pets**, since that was the AB3 behaviour. Now, you should change them to reflect the changes you have made!

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -28,7 +28,8 @@ public SortCommand(String field) {
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.sortPetList(field);
-        return new CommandResult(MESSAGE_SUCCESS);
+        model.updateFilteredPetList();
+        return new CommandResult(MESSAGE_SUCCESS + field);
```
Comment (soloplxya) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:32]: Not sure if this is the problem but I see that in `SortCommandTest.java`, the success message seems to be missing the `field`, which could be why the program is failing the sort test cases.

File: src/main/java/seedu/address/logic/commands/DietCommand.java [hunk 3]
```diff
@@ -59,7 +58,7 @@ public CommandResult execute(Model model) throws CommandException {
                 petToEdit.getAttendanceHashMap());
 
         model.setPet(petToEdit, editedPet);
-        model.updateFilteredPetList(PREDICATE_SHOW_ALL_PETS);
+        model.updateFilteredPetList();
```
Comment (soloplxya) [inline src/main/java/seedu/address/logic/commands/DietCommand.java:61]: Not too sure about this one as well. Maybe you can add a line `expectedModel.updateFilteredPetList()` after the ` expectedModel.setPet(firstPerson, editedPerson);` line as shown in the image below. 

image

**Pull request merged**